packages/api/src/modules/users/router.ts

```diff
@@ -0,0 +1,260 @@
+    const { id } = ctx.params;
+
+    const game = await getPBs(id);
+
+    if (!game) {
+        ctx.status = 400;
+        return (ctx.body = { message: "That user does not have any PBs!" });
+    }
+
+    ctx.status = 200;
+    ctx.body = game;
+
+    await next();
+});
+
+router.get("/userData/:id", async (ctx, next) => {
+    const { id } = ctx.params;
+
+    const data = await getUserData("id", id);
+
+    if (!data) {
+        ctx.status = 404;
+        return (ctx.body = { message: "That users doesn't exist." });
+    }
+
+    ctx.status = 200;
+    ctx.body = data;
+
+    await next();
+});
+
+router.get("/achievements/:id", async (ctx, next) => {
+    const { id } = ctx.params;
+
+    const achievements = await userAchievements(id);
+
+    if (achievements === null) {
+        throw new HttpError(400, "That user does not exist!");
+    }
+
+    ctx.status = 200;
+    ctx.body = achievements;
+    await next();
+});
+
+router.patch(
+    "/updateCountry",
+    requireAuthenticated(),
+    validateSchema(UpdateCountry, "body"),
+    async (ctx, next) => {
+        const { country } = ctx.request.body;
+        const { user } = ctx.session!;
+
+        await updateCountry("id", user, country);
+
+        ctx.status = 201;
+        ctx.body = {
+            message: "Successfully updated countrycode!"
+        };
+
+        await next();
+    }
+);
+
+router.patch(
+    "/updateDescription",
+    requireAuthenticated(),
+    validateSchema(UpdateCountry, "body"),
+    async (ctx, next) => {
+        const { description } = ctx.request.body;
+        const { user } = ctx.session!;
+
+        await updateDescription("id", user, description);
+
+        ctx.status = 200;
+        ctx.body = {
+            message: "Successfully updated description!"
+        };
+
+        await next();
+    }
+);
+
+router.patch("/changePassword", requireAuthenticated(), async (ctx, next) => {
+    const { user } = ctx.session!;
+
+    const { oldPassword, newPassword } = ctx.request.body;
+
+    const response = await changePassword(user, oldPassword, newPassword);
+
+    if (response) {
+        throw new HttpError(400, response);
+    }
+
+    ctx.status = 200;
+    ctx.body = { message: "Successfully changed password" };
+
+    await next();
+});
+
+router.post("/requestForgotPassword", async (ctx, next) => {
+    const { email } = ctx.request.body;
+
+    const emailExists = await forgotPassword(email);
+
+    if (!emailExists) {
+        throw new HttpError(400, "The provided email does not have an account");
+    }
+
+    ctx.status = 200;
+    ctx.body = {
+        message: "Success, an email has been sent to your email address"
+    };
+
+    await next();
+});
+
+router.get("/forgotPassword/:key", async (ctx, next) => {
+    const { key } = ctx.params;
+
+    if (await keyValid(key)) {
+        ctx.status = 200;
+        ctx.redirect(
+            `${process.env.SERVER_URL}/#/loginregister/resetPassword/${key}`
```

Comment:
Pretty sure that SERVER_URL is the url of the api and CORS_ORIGIN is the url of the front end.

Reply:
Yeah, that seems about right. I mistakenly put CORS_ORIGIN there, forget about it. Plus, it's a monorepo